lib/events.js

```diff
@@ -163,24 +161,21 @@ function emitMany(handler, isFn, self, args) {
   }
 }
 
-EventEmitter.prototype.emit = function emit(type) {
-  var er, handler, len, args, i, events, domain;
-  var needDomainExit = false;
-  var doError = (type === 'error');
+EventEmitter.prototype.emit = function emit(type, ...args) {
+  let doError = (type === 'error');
 
-  events = this._events;
-  if (events)
-    doError = (doError && events.error == null);
+  const events = this._events;
+  if (events !== undefined)
+    doError = (doError && events.error === undefined);
   else if (!doError)
     return false;
 
-  domain = this.domain;
+  const domain = this.domain;
 
   // If there is no 'error' event listener then throw.
   if (doError) {
-    if (arguments.length > 1)
-      er = arguments[1];
-    if (domain) {
+    let er = args[0];
```

Comment:
This is bad in case args is of length zero.

Reply:
Yep, thanks for catching it! That was never meant to make it into the PR because it's about 10x slower than checking length first.

Reply:
@bmeurer wouldn't it be a good idea to improve this in v8? It is somewhat weird that it is best to check for the length first instead of having v8 doing this automatically.

Reply:
Reading out of bounds is tricky in JS and requires some machinery. But yes we should probably fix this, esp. since jQuery and Co. already do that and don't want to give up on that practice. Can you file a bug on the V8 tracker pointing to this example?

cc @mathiasbynens